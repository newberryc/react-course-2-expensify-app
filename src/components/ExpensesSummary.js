import React from 'react';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import { connect } from 'react-redux';

export const ExpensesSummary = (props) => (
  <div>
    { props.expensesCount === 0 ? '' : `Viewing ${props.expensesCount} expense${props.expensesCount === 1 ? '' : 's'} totalling ${numeral(props.expensesTotal).format('$0,0.00')}`}
  </div>
);

const mapStateToProps = (state) => {
  const selectedExpenses = selectExpenses(state.expenses, state.filters);
  return {
      expensesCount: selectedExpenses.length,
      expensesTotal: selectExpensesTotal(selectedExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
