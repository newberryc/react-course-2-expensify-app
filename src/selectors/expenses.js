import moment from 'moment';

// visible expenses

export default (expenses, {text, sortBy, startDate, endDate}) => {
    return expenses.filter((expense) => {
        const createdAtMoment = moment(expense.createdAt);
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment,'day') : true;
        const textPattern = new RegExp(text, 'i');
        const textMatch = !(text && !textPattern.test(expense.description) && !textPattern.test(expense.note));
        return startDateMatch && endDateMatch && textMatch;
    }).sort((a, b) => {
        if (sortBy === 'date') {
            return a.createdAt < b.createdAt ? 1 : -1;
        } else if (sortBy === 'amount') {
            return a.amount < b.amount ? 1 : -1;
        }
    });
};
