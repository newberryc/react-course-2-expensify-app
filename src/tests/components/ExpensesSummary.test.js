import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render summary for single expense', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={expenses[0].amount / 100} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render summary for multiple expenses', () => {
  const expensesTotal = (expenses[0].amount + expenses[1].amount + expenses[2].amount) / 100;
  const wrapper = shallow(<ExpensesSummary expensesCount={expenses.length} expensesTotal={expensesTotal} />);
  expect(wrapper).toMatchSnapshot();
});
