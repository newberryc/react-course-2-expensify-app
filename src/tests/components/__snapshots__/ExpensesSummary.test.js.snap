// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render summary for multiple expenses 1`] = `
<div>
  Viewing 3 expenses totalling $1,996.95
</div>
`;

exports[`should render summary for single expense 1`] = `
<div>
  Viewing 1 expense totalling $1.95
</div>
`;
